packages/engine/src/store/selectors/getSketchParamValues.ts

```diff
@@ -0,0 +1,27 @@
+import { EngineState } from '@store/types'
+
+// Get the values of the parameters of a sketch, dealing with child nodes
+export const getSketchParamValues = (state: EngineState, sketchId: string) => {
+  const { sketches, nodeValues, nodes } = state
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const paramValues: { [key: string]: any } = {}
+  const sketch = sketches[sketchId]
+
+  sketch.paramIds.forEach((id) => {
+    const { key, valueType } = nodes[id]
+
+    let value
+
+    if (valueType === 'vector3') {
+      // Return an array of values for vector3
+      const childNodeIds = nodes[id].childNodeIds
+      value = childNodeIds.map((childNodeId) => nodeValues[childNodeId])
```

Comment:
We're getting all child nodes and creating an array to be handed to the sketch. Is that how users will want it? We could also hand it back as an object with `x`, `y` and `z` properties (or `r`, `g`, `b` for colors...). Part of me thinks an array is simplest and the user can deal with it how they please.